src/koans/array.rs

```diff
@@ -6,11 +6,76 @@ fn array_index() {
   assert!(__ == 1);
 }
 
-// A new fixed size array can be created by declaring
-// the type of its elements along with its length
+// A new fixed size array can be created by declaring the type of its elements
+// along with its length
 // [i32; 0] = []
 #[test]
 fn array_empty() {
   let arr: __;
   assert!(arr.len() == 0);
 }
+
+// Elements can be replaced in an array at a certain index.
+// hint: Without the 'mut' keyword, you won't be able to change data.
+#[test]
+fn insert_at_index() {
+  let mut arr: [u8; 5] = [0, 1, 2, 3, 4];
+  __ = 0;
+  assert!(arr == [0, 1, 2, 3, 0]);
+}
+
+// Arrays can be iterated over.
+#[test]
+fn array_iteration() {
+  let arr: [u8; 3] = [3, 2, 1];
+  let mut iterator = arr.iter();
+  assert!(iterator.next().unwrap() == &__);
+  assert!(iterator.next().unwrap() == &__);
+  assert!(iterator.next().unwrap() == &__);
+}
+
+// Arrays can also be mutated during iteration
+#[test]
+fn array_map() {
+  let arr: [u32; 4] = [2, 5, 7, 4];
+  let mut iterator = arr.iter().map(__);
+  assert!(iterator.next().unwrap() == 4);
```

Comment:
Actually I just realized since that doesn't need to be filled in, it's totally fine to write it that way.